wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl

```diff
@@ -0,0 +1,3586 @@
+            help: "This factor will be multiplied to the size of TAG-ALIGNs (BEDs) to determine required disk size of instance on GCP/AWS.",
+        }
+        macs2_signal_track_mem_factor: {
+            description: "Multiplication factor to determine memory required for task macs2_signal_track.",
+            group: "resource_parameter",
+            help: "This factor will be multiplied to the size of TAG-ALIGNs (BEDs) to determine required memory of instance (GCP/AWS) or job (HPCs).",
+        }
+        macs2_signal_track_time_hr: {
+            description: "Walltime (h) required for task macs2_signal_track.",
+            group: "resource_parameter",
+            help: "This is for HPCs only. e.g. SLURM, SGE, ...",
+        }
+        macs2_signal_track_disk_factor: {
+            description: "Multiplication factor to determine persistent disk size for task macs2_signal_track.",
+            group: "resource_parameter",
+            help: "This factor will be multiplied to the size of TAG-ALIGNs (BEDs) to determine required disk size of instance on GCP/AWS.",
+        }
+        align_trimmomatic_java_heap: {
+            description: "Maximum Java heap (java -Xmx) in task align.",
+            group: "resource_parameter",
+            help: "Maximum memory for Trimmomatic. If not defined, 90% of align task's memory will be used.",
+        }
+        filter_picard_java_heap: {
+            description: "Maximum Java heap (java -Xmx) in task filter.",
+            group: "resource_parameter",
+            help: "Maximum memory for Picard tools MarkDuplicates. If not defined, 90% of filter task's memory will be used.",
+        }
+        gc_bias_picard_java_heap: {
+            description: "Maximum Java heap (java -Xmx) in task gc_bias.",
+            group: "resource_parameter",
+            help: "Maximum memory for Picard tools CollectGcBiasMetrics. If not defined, 90% of gc_bias task's memory will be used.",
+        }
+    }
+
+    input {
+        # group: runtime_environment
+        String docker = "encodedcc/chip-seq-pipeline:v2.2.2"
+        String singularity = "https://encode-pipeline-singularity-image.s3.us-west-2.amazonaws.com/chip-seq-pipeline_v2.2.2.sif"
+        String conda = "encd-chip"
+        String conda_macs2 = "encd-chip-macs2"
+        String conda_spp = "encd-chip-spp"
+
+        # group: pipeline_metadata
+        String title = "Untitled"
+        String description = "No description"
+
+        # group: reference_genome
+        File? genome_tsv
+        String? genome_name
+        File? ref_fa
+        File? bwa_idx_tar
+        File? bowtie2_idx_tar
+        File? chrsz
+        File? blacklist
+        File? blacklist2
+        String? mito_chr_name
+        String? regex_bfilt_peak_chr_name
+        String? gensz
+        File? custom_aligner_idx_tar
+
+        # group: input_genomic_data
+        Boolean? paired_end
+        Array[Boolean] paired_ends = [
+        ]
```

Comment:
Should this be line split?

Reply:
There's currently no logic for detecting empty braces/brackets/etc but it would be easy to add, so sure. The current logic is [incorrectly] assuming non-empty arrays which should have one element per line. Although you and I have discussed how that rule often has exceptions which the formatter does not recognize.

Open to suggestions for how to handle arrays